WordPress/Classes/Services/ReaderTopicService.m

```diff
@@ -362,6 +366,48 @@ - (ReaderTopic *)siteTopicForPost:(ReaderPost *)post
     return topic;
 }
 
+- (void)siteTopicForSiteWithID:(NSNumber *)siteID
+                       success:(void (^)(NSManagedObjectID *objectID, BOOL isFollowing))success
+                       failure:(void (^)(NSError *error))failure
+{
+    ReaderTopicServiceRemote *remoteService = [[ReaderTopicServiceRemote alloc] initWithApi:[self apiForRequest]];
+    [remoteService fetchSiteInfoForSiteWithID:siteID success:^(RemoteReaderSiteInfo *siteInfo) {
+        if (!success) {
+            return;
+        }
+        NSEntityDescription *entity = [NSEntityDescription entityForName:NSStringFromClass([ReaderTopic class])
+                                                  inManagedObjectContext:self.managedObjectContext];
+        ReaderTopic *topic = [[ReaderTopic alloc] initWithEntity:entity
+                     insertIntoManagedObjectContext:self.managedObjectContext];
```

Comment:
Just in case: This is *always* gonna be running using the Main Context right? (otherwise we'd need the `performBlock` methods)